assets/src/edit-story/components/panels/textStyle.js

```diff
@@ -0,0 +1,287 @@
+  const lineHeight = getCommonValue(selectedElements, 'lineHeight');
+  const padding = getCommonValue(selectedElements, 'padding') || '';
+  const fontFamily = getCommonValue(selectedElements, 'fontFamily');
+  const fontSize = getCommonValue(selectedElements, 'fontSize');
+  const fontWeight = getCommonValue(selectedElements, 'fontWeight');
+  const fontWeights = getCommonValue(selectedElements, 'fontWeights');
+  const fontStyle = getCommonValue(selectedElements, 'fontStyle');
+  const fontFallback = getCommonValue(selectedElements, 'fontFallback');
+
+  const {
+    state: { fonts },
+    actions: { getFontWeight, getFontFallback },
+  } = useFont();
+  const [state, setState] = useState({
+    fontFamily,
+    fontStyle,
+    fontSize,
+    fontWeight,
+    fontFallback,
+    fontWeights,
+    textAlign,
+    letterSpacing,
+    lineHeight,
+    padding,
+  });
+  const [lockRatio, setLockRatio] = useState(true);
+  useEffect(() => {
+    setState({ textAlign, letterSpacing, lineHeight, padding });
+  }, [textAlign, letterSpacing, lineHeight, padding]);
+  useEffect(() => {
+    const currentFontWeights = getFontWeight(fontFamily);
+    const currentFontFallback = getFontFallback(fontFamily);
+    setState({
+      fontFamily,
+      fontStyle,
+      fontSize,
+      fontWeight,
+      fontWeights: currentFontWeights,
+      fontFallback: currentFontFallback,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [fontFamily, fontStyle, fontSize, fontWeight, getFontWeight]);
+  const handleSubmit = (evt) => {
+    onSetProperties((properties) => {
+      const { width, height: oldHeight, rotationAngle, x, y } = properties;
+      const updatedState = removeUnsetValues(state);
+      const newProperties = { ...properties, ...updatedState };
+      const newHeight = dataPixels(calculateTextHeight(newProperties, width));
+      const [dx, dy] = calcRotatedResizeOffset(
+        rotationAngle,
+        0,
+        0,
+        0,
+        newHeight - oldHeight
+      );
+      return {
+        ...updatedState,
+        height: newHeight,
+        x: dataPixels(x + dx),
+        y: dataPixels(y + dy),
+      };
+    });
+
+    evt.preventDefault();
+  };
+
+  const alignmentOptions = [
+    { name: __('Default', 'web-stories'), value: '' },
+    { name: __('Left', 'web-stories'), value: 'left' },
+    { name: __('Right', 'web-stories'), value: 'right' },
+    { name: __('Center', 'web-stories'), value: 'center' },
+    { name: __('Justify', 'web-stories'), value: 'justify' },
+  ];
+
+  const fontStyles = [
+    { name: __('Normal', 'web-stories'), value: 'normal' },
+    { name: __('Italic', 'web-stories'), value: 'italic' },
+  ];
+
+  return (
+    <SimplePanel
+      name="style"
+      title={__('Style', 'web-stories')}
+      onSubmit={handleSubmit}
+    >
+      <Row>
+        {fonts && (
+          <SelectMenu
+            ariaLabel={__('Font family', 'web-stories')}
```

Comment:
I see `ariaLabel` being used a lot, but none of the components actually support it.

For example, `SelectMenu` does not have any `ariaLabel` prop defined.

Reply:
Good points! aria-label added to input and selectmenu